processor/processorhelper/metrics_test.go

```diff
@@ -67,3 +74,78 @@ func newTestMProcessor(retError error) ProcessMetricsFunc {
 		return md, retError
 	}
 }
+
+func TestMetricsProcessor_RecordInOut(t *testing.T) {
+	// Regardless of how many data points are ingested, emit just one
+	mockAggregate := func(_ context.Context, _ pmetric.Metrics) (pmetric.Metrics, error) {
+		md := pmetric.NewMetrics()
+		md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetEmptySum().DataPoints().AppendEmpty()
+		return md, nil
+	}
+
+	incomingMetrics := pmetric.NewMetrics()
+	dps := incomingMetrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetEmptySum().DataPoints()
+
+	// Add 3 data points to the incoming
+	dps.AppendEmpty()
+	dps.AppendEmpty()
+	dps.AppendEmpty()
+
+	metricReader := sdkmetric.NewManualReader()
```

Comment:
Could the test code here use the generated test code? Can be done in a follow up PR